plugin/v3/plugin.proto

```diff
@@ -102,6 +109,38 @@ message Read {
   }
 }
 
+message TableRelation {
+  string table_name = 1;
+  string parent_table = 2;
+}
+
+
+
+message Predicate {
+  enum Operator {
+    EQ = 0;
+    // LT = 1;
+    // LTE = 2;
+    // GT = 3;  
+    // GTE = 4;
+  }
+
+  Operator operator = 1;
+  string column = 2;
+  // marshalled arrow.Record
+  bytes record = 3;
+}
+
+message PredicatesGroup {
+  enum GroupingType {
+    AND = 0;
+    OR = 1;
+  }
+  GroupingType grouping_type = 1;
+	repeated Predicate predicates = 2;
```

Comment:
I feel like this is an overkill.

Instead, we need to supply column list + record to compare against.
This way, if the column list is empty, we just need to compare whole record, but if it isn'n then we just compare the values to the ones from the supplied single record.

(now you supply record per column, I suggest supplying single record (=row) & just indicate what columns are required for comparison)

Reply:
IMO we should just use expression syntax, like:
```antrl4
Expression : UnaryExpression | BinaryExpression
UnaryExpression: Not Expression | value
BinaryExpression: UnaryExpression cmp UnaryExpression | UnaryExpression AND UnaryExpression | UnaryExpression OR UnaryExpression
```

Reply:
Example: https://github.com/cloudquery/plugin-pb/pull/19